js/ding_availability_labels.js

```diff
@@ -182,16 +173,32 @@
        *   jQuery availability element to add the class to.
        * @param btn
        *   Reservation button to add the class to.
-       * @param class_name
-       *   The class to add to the elements.
+       * @param status
+       *   Structure with available and reservable state.
        */
-      function update_availability_elements(element, btn, class_name) {
-        element.addClass(class_name);
-        if (btn.length) {
-          btn.addClass(class_name);
+      function update_availability_elements(element, btn, status) {
+        var class_name = null;
+
+        for (var i in status) {
+          if (status[i] === true) {
+            class_name = i;
+          }
+          else {
+            if (i === 'available') {
+              class_name = 'un' + i;
+            }
+            else if (i === 'reservable') {
+              class_name = 'not-' + i;
+            }
+          }
+
+          element.addClass(class_name);
+          if (btn.length) {
+            btn.addClass(class_name);
+          }
```

Comment:
Well done!

Is this none usable in [`ding_availability.js`](https://github.com/ding2/ding_availability/pull/15/files#diff-c7980ed510e19f8dc6f2b06c892ee392R101)? It seems as if it addresses the exact same things.